src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -149,20 +149,41 @@ std::shared_ptr<Table> SyntheticTableGenerator::generate_table(
           values.push_back(generate_value_by_distribution_type());
         }
 
-        segments.push_back(
-            std::make_shared<ValueSegment<ColumnDataType>>(create_typed_segment_values<ColumnDataType>(values)));
+        segments[column_index] =
+            std::make_shared<ValueSegment<ColumnDataType>>(create_typed_segment_values<ColumnDataType>(values));
       });
+
       // add full chunk to table
       if (column_index == num_columns - 1) {
-        const auto mvcc_data = std::make_shared<MvccData>(segments.front()->size(), CommitID{0});
-        table->append_chunk(segments, mvcc_data);
+        if (use_mvcc == UseMvcc::Yes) {
+          const auto mvcc_data = std::make_shared<MvccData>(segments.front()->size(), CommitID{0});
+          table->append_chunk(segments, mvcc_data);
+        } else {
+          table->append_chunk(segments);
+        }
+
+        if (segment_encoding_specs) {
+          /**
+          * For chunk sizes <1M, parallelizing the segment creation is not worth the overhead, because chunk encoding 
+          * dominates the runtime. Nonetheless, a single thread for the chunk encoder (which itself is parallelizing
+          * internally) ensures that the segment creation is not blocked by the encoding of the previous chunk. To
+          * avoid overparallelizing, there is at most one thread for encode_chunk() at any time.
+          **/
+          if (!encoder_threads.empty()) {
+            encoder_threads.back().join();
```

Comment:
I don't really get this part. In which case will there be more than one encoder_thread?

Reply:
There will never be, we just wait for the last thread (`.back()`) to finish.
Since the data generation is much faster than encoding, we would create many more encoders if we don't wait for the previous one.
I failed using a single std::thread variable and reusing it, thus the vector.

Reply:
As I totally see your argument to avoid `std::thread` in `src/lib`, I think I'll call`encode_segment()` via some `std::threads` (duplicating 5 lines of marking the chunk as immutable).
This way, we don't have to set/reset the scheduler and we don't introduce any manual threads to core Hyrise. What do you think?